hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/sftp/TestSFTPFileSystem.java

```diff
@@ -374,4 +374,13 @@ public void testMkDirs() throws IOException {
     assertThat(((SFTPFileSystem) sftpFs).getConnectionPool().getLiveConnCount(),
         is(1));
   }
+
+  @Test
+  public void testCloseFileSystemClosesConnectionPool() throws Exception {
+    SFTPFileSystem fs = (SFTPFileSystem) sftpFs;
+    fs.getHomeDirectory();
+    assertThat(fs.getConnectionPool().getLiveConnCount(), is(1));
+    fs.close();
+    assertThat(fs.getConnectionPool().getLiveConnCount(), is(0));
+  }
```

Comment:
can you add another close() here just so we can make sure that re-entrant close calls are safe.

Reply:
Hey Steve, thanks a lot for reviewing, I appreciate it.

A FileSystem instance already gets closed after each test method, [please take a look over here](https://github.com/apache/hadoop/blob/ca2e5a462daeb48f2e18e6a0ebe2fd6a3f20e89e/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/sftp/TestSFTPFileSystem.java#L117). I hope this makes the case you pointed out covered. Please, let me know if it doesn't, I will add an extra call to make it look more explicit.

Reply:
we swallow those exceptions in teardown as otherwise their exceptions get reported over something in the test case itself (or setup). Please add the extra call for completeness. thanks

Reply:
exactly, I missed the catch block, thank you! Done.